### PR TITLE
BGDIINF_SB-1534: avoid duplicate validation in items

### PR DESCRIPTION
**Pull request by ltshb**
The item properties datetimes were validated twice in a row. This was
unecessary and due to a wrong uses of the django/rest_framework in
unittest. Creating an object with
ModelClass.objects.create(**validated_data) should
only be done with validated data, like for example in the serializer.
This method doesn't do any validation. If we don't have validated data,
then we need to use ModelClass(**unvalidated_data).full_clean().save()

So the item model unittest has been corrected in order to remove the
self.clean() in the item.save() method. This introduce a sligthly
performance improvement, but more important it makes the code cleaner.

### CODE REVIEW

**Review by hansmannj — APPROVED** [review]
Looks good to me :+1:

**Review by boecklic — APPROVED** [review]

File: app/stac_api/serializers.py [hunk 1]
```diff
@@ -735,12 +735,21 @@ def update(self, instance, validated_data):
...
-            partial=self.partial
-        )
+        if (
+            not self.partial or \
+            'properties_datetime' in attrs or \
+            'properties_start_datetime' in attrs or \
+            'properties_end_datetime' in attrs
+        ):
+            validate_item_properties_datetimes(
+                attrs.get('properties_datetime', None),
```
Comment (procrastinatio) [inline app/stac_api/serializers.py:745]: So `None`is a valid date?

**Review by procrastinatio — COMMENTED** [review]
Would also test if `date`is `None`

File: app/stac_api/serializers.py (hunk 1 above)
Comment (ltshb) [reply app/stac_api/serializers.py:745]: None is not a valide date, but in properties there is 3 dates:
- properties_datetime
- properties_start_datetime
- properties_end_datetime

and either the first one is required or both start and end datetime, that's what validate the function.
